referenceframe/urdf.go

```diff
@@ -0,0 +1,313 @@
+		}
+
+		mc.Links = append(mc.Links, childLink)
+	}
+
+	// Handle links
+	for _, linkElem := range urdf.Links {
+		// Skip any world links
+		if linkElem.Name == "world" {
+			continue
+		}
+
+		// Find matching links which already exist, take care of geometry if collision elements are detected
+		hasCollision := len(linkElem.Collision) > 0
+		for idx, prefabLink := range mc.Links {
+			if prefabLink.ID == linkElem.Name && hasCollision {
+				geoCfg, _ := createConfigFromCollision(linkElem)
+				mc.Links[idx].Geometry = geoCfg
+				break
+			}
+		}
+
+		// In the event the link does not already exist in the ModelConfig, we will have to generate it now
+		// Most likely, this is a link normally whose parent is the World
+		if _, ok := parentMap[linkElem.Name]; !ok {
+			thisLink := JsonLink{ID: linkElem.Name, Parent: World}
+			thisLink.Translation = spatial.TranslationConfig{0.0, 0.0, 0.0}
+			thisLink.Orientation = spatial.OrientationConfig{} // Orientation is guaranteed to be zero for this
+
+			if err != nil {
+				return nil, err
+			}
+
+			if hasCollision {
+				geoCfg, _ := createConfigFromCollision(linkElem)
+				thisLink.Geometry = geoCfg
+			}
+
+			mc.Links = append(mc.Links, thisLink)
+		}
+	}
+
+	return mc, nil
+}
+
+// Convenience method to split up space-delimited fields in URDFs, such as xyz or rpy attributes.
+func convStringAttrToFloats(attr string) []float64 {
+	var converted []float64
+	attr_slice := strings.Fields(attr)
+
+	for _, value := range attr_slice {
+		value, _ := strconv.ParseFloat(value, 64)
+		converted = append(converted, value)
+	}
+
+	return converted
+}
+
+// Convenience method to simplify creating geometry configs from URDF XML that has a collision element specified
+func createConfigFromCollision(link UrdfLink) (spatial.GeometryConfig, error) {
+	var geoCfg spatial.GeometryConfig
+	boxGeometry := link.Collision[0].Geometry.Box
+	sphereGeometry := link.Collision[0].Geometry.Sphere
+
+	// Offset for the geometry origin from the reference link origin
+	geomXYZ := convStringAttrToFloats(link.Collision[0].Origin.XYZ)
+	geomTx := spatial.TranslationConfig{geomXYZ[0], geomXYZ[1], geomXYZ[2]}
+	geomRPY := convStringAttrToFloats(link.Collision[0].Origin.RPY)
+	geomEA := spatial.EulerAngles{
+		Roll:  utils.RadToDeg(geomRPY[0]),
+		Pitch: utils.RadToDeg(geomRPY[1]),
+		Yaw:   utils.RadToDeg(geomRPY[2]),
+	}
+	geomOx, err := spatial.NewOrientationConfig(geomEA.AxisAngles())
+
+	if err != nil {
+		return spatial.GeometryConfig{}, err
+	}
+
+	// Logic specific to the geometry type
+	if len(boxGeometry.Size) > 0 {
+		boxDims := convStringAttrToFloats(boxGeometry.Size)
+		geoCfg = spatial.GeometryConfig{
+			Type:              "box",
+			X:                 metersToMM(boxDims[0]),
+			Y:                 metersToMM(boxDims[1]),
+			Z:                 metersToMM(boxDims[2]),
+			TranslationOffset: geomTx,
+			OrientationOffset: *geomOx,
+			Label:             "box",
+		}
+	} else if sphereGeometry.Radius > 0 {
```

Comment:
The way the geometry config parser is written this should be a lot simpler to do.  You should be able to just shove everything in the same config struct and have it figure it out, since this logic is replicated there

Reply:
Yeah, I'm not going to mess with that at the moment, though.